counterpartylib/server.py

```diff
@@ -0,0 +1,350 @@
+#! /usr/bin/env python3
+
+import os
+import decimal
+import sys
+import logging
+logger = logging.getLogger(__name__)
+import time
+import dateutil.parser
+import calendar
+import traceback
+import binascii
+import socket
+import signal
+import appdirs
+import platform
+
+from counterpartylib.lib import api, config, util, exceptions, blocks, check, backend, database, transaction, script, log
+
+D = decimal.Decimal
+
+
+class ConfigurationError(Exception):
+    pass
+
+
+def sigterm_handler(_signo, _stack_frame):
+    if _signo == 15:
+        signal_name = 'SIGTERM'
+    elif _signo == 2:
+        signal_name = 'SIGINT'
+    else:
+        assert False
+    logger.info('Received {}.'.format(signal_name))
+
+    if 'api_server' in globals():
+        logger.info('Stopping API server.')
+        api_server.stop()
+        api_status_poller.stop()
+    logger.info('Shutting down.')
+    logging.shutdown()
+    sys.exit(0)
+signal.signal(signal.SIGTERM, sigterm_handler)
+signal.signal(signal.SIGINT, sigterm_handler)
+
+
+# Lock database access by opening a socket.
+class LockingError(Exception):
+    pass
+def get_lock():
+    logger.info('Acquiring lock.')
+
+    # Cross‐platform.
+    if os.name == 'nt' or platform.system() == 'Darwin':    # Windows or OS X
+        # Not database‐specific.
+        socket_family = socket.AF_INET
+        socket_address = ('localhost', 8999)
```

Comment:
Good question, but realistically, how many Windows and Mac users are doing something else on port 8999 (that won't patch the code themselves)? (If people complain, we can always add the configuration parameter in later.)